lib/Doctrine/DBAL/Types/DateIntervalType.php

```diff
@@ -25,7 +26,7 @@ public function getName()
      */
     public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
     {
-        $fieldDeclaration['length'] = 255;
+        $fieldDeclaration['length'] = min($platform->getCharMaxLength(), 255);
         $fieldDeclaration['fixed']  = true;
```

Comment:
According to the format, the interval is definitely not a 255-char fixed length string.

@Majkl578 would `CHAR(255)` → `VARCHAR(32)` be a breaking change given that the current type is far from truth? `VARCHAR(128)` still should be safe and portable according to https://github.com/doctrine/dbal/pull/2316/files#r53146155.